step-api/src/main/java/org/jenkinsci/plugins/workflow/structs/DescribableHelper.java

```diff
@@ -419,6 +434,7 @@ public Schema getType() {
         private final Class<?> supertype;
         private final Map<String,Schema> types;
         HeterogeneousObjectType(Class<?> supertype, Map<String,Schema> types) {
+            super(supertype);
             this.supertype = supertype;
```

Comment:
I meant that `HeterogeneousObjectType.supertype` could be deleted, with a typesafe getter retained like

```java
public Class<?> getSupertype() {
    return (Class) getType(); // from superclass
}
```